pkg/cmd/roachtest/tests/admission_control_index_overload_tpce.go

```diff
@@ -0,0 +1,172 @@
+			defer cleanupFunc()
+		}
+
+		t.Status("installing docker")
+		if err := c.Install(ctx, t.L(), workloadNode, "docker"); err != nil {
+			t.Fatal(err)
+		}
+
+		importC := make(chan struct{})
+		m := c.NewMonitor(ctx, roachNodes)
+		m.Go(func(ctx context.Context) error {
+			const dockerRun = `sudo docker run cockroachdb/tpc-e:latest`
+
+			roachNodeIPs, err := c.InternalIP(ctx, t.L(), roachNodes)
+			if err != nil {
+				return err
+			}
+			roachNodeIPFlags := make([]string, len(roachNodeIPs))
+			for i, ip := range roachNodeIPs {
+				roachNodeIPFlags[i] = fmt.Sprintf("--hosts=%s", ip)
+			}
+
+			t.Status("preparing workload (~4 hours)")
+			c.Run(ctx, workloadNode, fmt.Sprintf("%s --customers=%d --racks=%d --init %s",
+				dockerRun, opts.customers, racks, roachNodeIPFlags[0]))
+			close(importC)
+
+			t.Status("running workload (~4 hours)")
+			activeCustomers := int(float64(opts.customers) * opts.activeRatio)
+			result, err := c.RunWithDetailsSingleNode(ctx, t.L(), workloadNode,
+				fmt.Sprintf("%s --customers=%d --active-customers=%d --racks=%d --duration=%s --threads=%d %s",
+					dockerRun, opts.customers, activeCustomers, racks, runDuration, opts.threads, strings.Join(roachNodeIPFlags, " ")))
+			if err != nil {
+				t.Fatal(err.Error())
+			}
+			t.L().Printf("workload output:\n%s\n", result.Stdout)
+			if strings.Contains(result.Stdout, "Reported tpsE :    --   (not between 80% and 100%)") {
+				return errors.New("invalid tpsE fraction")
+			}
+			return nil
+		})
+		m.Go(func(ctx context.Context) error {
+			// Wait for the import to complete.
+			t.Status("waiting for import to complete")
+			select {
+			case <-importC:
+			case <-ctx.Done():
+				return ctx.Err()
+			}
+
+			t.Status("waiting before creating the index")
+
+			// Wait for the load to run for a while before starting the index build.
+			select {
+			case <-time.After(opts.delayBeforeIndex):
+			case <-ctx.Done():
+				return ctx.Err()
+			}
+
+			t.Status("creating index (~30min)")
+
+			db := c.Conn(ctx, t.L(), 1)
+			_, err := db.ExecContext(ctx, "CREATE INDEX ON tpce.cash_transaction (ct_dts)")
+			if err != nil {
+				_ = db.Close()
+				return err
+			}
+
+			t.Status("index creation complete")
+
+			return db.Close()
+		})
+		m.Wait()
+	}
+
+	// This test takes about 8 hours to run on 10 N2 machines.
+	r.Add(registry.TestSpec{
+		Name:    "admission-control/index-overload-tpce",
+		Owner:   registry.OwnerAdmissionControl,
+		Cluster: spec.MakeClusterSpec(spec.GCE, "n2-standard-8", "Intel Ice Lake", 10, spec.CPU(8), spec.SSD(1)),
+		Tags:    []string{`weekly`},
+		Run: func(ctx context.Context, t test.Test, c cluster.Cluster) {
+			opts := tpceOptions{
+				customers:        100_000,
+				activeRatio:      0.2,
+				threads:          400,
+				delayBeforeIndex: 1 * time.Hour,
+			}
+			runTPCE(ctx, t, c, opts)
+		},
+	})
+
+	// This test takes about 1 hours to run.
+	r.Add(registry.TestSpec{
```

Comment:
Meh, let's just have the one test above and make sure it runs <1h. And also, within reason, support either --local and/or --skip-init if you use an already set up cluster to keep re-running things without the bulk import. This TPC-E harness also needs to integrate with prometheus/grafana (I see you've done so manually above), let's do that in this PR -- steal as many ideas from the TPC-C harness when doing so.

Reply:
I'm not sure I understand this request. I don't think it is possible to create a test that runs <1hr, uses default settings and sees this overload issue (since it needs 1000 L0 files as part of index creation). I wasn't sure how best to approach this, so definitely open to ideas on how to do this (combination of skipped tests, short test that doesn't show the issue or other).

Reply:
Removed the small test below.